samples/account-linking/csharp/README.md

```diff
@@ -24,6 +24,99 @@ Please see the [Code Tours](#code-tours) section for in-depth explanation of the
     After the user installs the app, they will be prompted when using any of the capabilities (tab / conversational bot / messaging extension) to log into GitHub. After logging in the app will store the users access and refresh tokens so that they won't need to log in again across the other capabilities or on other devices. 
 
     There are example in-memory and Azure implementations of the token persistance. 
+    
+You will find the implementation details below. 
+
+This sample will help you
+•	Show you the architecture to build out SSO in Teams
+•	Use this code (if you build in C#) 
+
+### Scope
+SSO is a problem across multiple surface areas and this sample can be used to build solutions in three surface areas:
+•	Tabs
+•	Bots
+•	Messaging extensions
+This sample does not need to cover:
+•	It does need to cover Graph tokens. These tokens are received by exchanging user access tokens on the server-side.
+•	It also does not need to be implemented in all languages, it can be implemented in one language to show the basic framework
+•	Specific auth system systems on the partner side -> the partner can form these connections youselves
+
+### Context
+#### Problem Overview
+AAD SSO is important for three parties:
+•	End user: Skip signing into apps in Teams
+•	Developer: Simpler development when it comes to authentication
+•	IT Admin: Protect user privacy and security through conditional access policies
+Often, you (a partner) have your own Identity and Access Management systems and your own user account information. In order for you to build out AAD SSO inside of Teams, you need to associate the users’ accounts with AAD SSO accounts. 
+
+#### SSO Overview
+Teams enacts AADD SSO in the following manner:
+1.	In the tab, call getAuthToken() using the Teams JavaScript SDK. This tells Teams to obtain an auth token.
+2.	If this is the first time the current user has used your tab application, you will be prompted to consent.
+3.	Microsoft Teams requests a token for the tab application from the Azure AD endpoint for the current user.
+4.	Azure AD sends the tab application token back to Teams.
+5.	Microsoft Teams then sends the token back to the tab application.
+6.	JavaScript in the tab application can parse the token and extract the information it needs, such as the user's email address. The tab app can optionally exchange the token server-side for further Graph permissions.
+
+### Problem Definition
+You have two approaches to link AAD accounts with existing customer accounts. These two scenarios are:
+1.	the user exists in the third-party app with the same email address
+2.	the user exists in the third-party app with a different email address
+3.	the user logs  in with Microsoft SSO and then provides consent for AAD to access partner’s account information
```

Comment:
I'm not sure this is really the problem definition. The problem is that:
1. You want the seamless experience Teams SSO provides
2. Your app doesn't use AzureAd so it can't use Teams SSO